crates/pipeline_manager/src/compiler.rs

```diff
@@ -140,7 +141,7 @@ impl Compiler {
                     if let Some(job) = &job {
                         // Program was deleted, updated or the user changed its status
                         // to cancelled -- abort compilation.
-                        let descr = db.lock().await.get_program_if_exists(job.program_id).await?;
+                        let descr = db.lock().await.get_program_if_exists(job.tenant_id, job.program_id).await?;
```

Comment:
Yeah, I did this for now because otherwise we'd need to change `CompilationJob` to have the entire `PipelineDescr` in it. We'll refactor this code out heavily when we have a separate compiler service anyway, so I'll save such a change for later.